python/lsst/cell_coadds/_fits.py

```diff
@@ -19,14 +19,57 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+"""Module to handle FITS serialization and de-serialization.
+
+The routines to write and read the files are in the same module, as change to
+one is typically accompanied by a corresponding change to another.
+Changes to this module may require a bump to the version number denoted by the
+module constant FILE_FORMAT_VERSION. The policy for bumping the version is:
+
+1. When the on-disk format written by this module changes such that the
+accompanying reader can still read files written by the previous version, then
+there should be a minor bump.
+
+2. When the on-disk format written by this module changes in a way that will
+prevent the reader from reading a previous version of the code, then there
+should be a major bump. This holds even if the reader temporarily supports a
+previous version with deprecation warnings.
+
+For example, a file with VERSION=1.3 should still be readable by the reader in
+this module when the module-level constant FILE_FORMAT_VERSION=1.4. A file
+written with VERSION=1.4 may also be readable by a reader when the module-level
+FILE_FORMAT_VERSION=1.3, although such a use case is not expected and may be
+only supported on a best-effort basis. A concrete example of change that
+requires only a minor bump is adding another BinTable that keeps tract of the
+input visits. An example of major change would be migrating from using
+BinTableHDU to ImageHDU to save data. Even if the reader supports reading
+either of this formats based on the value of VERSION from the header, it should
+be a major change. A major bump must also be recorded in the `isCompatibleWith`
+method.
+
+Unit tests only check that a file written can be read by the concurrent version
+of the module, but not by any of the previous ones. Hence, bumping
+FILE_FORMAT_VERSION to the appropriate value is ultimately at the discretion of
+the developers.
+
+It is plausible that different (non-consequent) major versions are compatible
+with each other (due to reverting back to an earlier format, or to something
+very similar). Such cases should be marked in the `isCompatibleWith` method.
```

Comment:
I would describe `isCompatibleWith` as testing whether a particular format version can be read by the current reader (such that the answer may be "yes" due to intentional backwards-compatibility).  It is not a measure of compatibility between formats.